indico/modules/events/sessions/clone.py

```diff
@@ -51,30 +58,43 @@ def _clone_sessions(self, new_event):
                           subqueryload('acl_entries')))
         for old_sess in query:
             sess = Session()
-            sess.populate_from_attrs(old_sess, attrs)
-            sess.blocks = list(self._clone_session_blocks(old_sess.blocks))
+            if event_exists:
+                sess.populate_from_attrs(old_sess, attrs, skip='friendly_id')
+                sess.friendly_id = self._get_next_friendly_id(new_event.id)
```

Comment:
⚠️ I already added some more comments around these lines, with the stuff in here most of them are probably obsolete.

Why do we need this? If you (actually) skip populating, the friendly IDs should be auto-assigned when the sessions get inserted.

Also, if there are any events we assume there's a conflict and do not allow importing. So in principle none of the special friendly-id handling should be needed. If you get an error due to non-unique friendly IDs it means we have a bug in our model and do not exclude deleted rows from the uniqueness check.

```python
    __auto_table_args = (db.Index(None, 'friendly_id', 'event_id', unique=True),
                         {'schema': 'events'})
```

should probably be

```python
    __auto_table_args = (db.Index(None, 'friendly_id', 'event_id', unique=True,
                                  postgresql_where=db.text('NOT is_deleted')),
                         {'schema': 'events'})
```

Reply:
Yes, there was a friendly_id uniqueness issue. I didn't think of that solution, but it should be much better this way.